### PR TITLE
TRUNK-6277: Migrate HL7InError from Hibernate Mapping XML to JPA annotations.

### PR DESCRIPTION
**Pull request by wikumChamith**
## Issue I worked on

see https://openmrs.atlassian.net/browse/TRUNK-6277

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by wikumChamith** [comment]
@dkayiwa, @ibacher I am not seeing this test error locally. Any suggestions on what might be causing this?

**Comment by ibacher** [comment]
There's some class you need to register the annotated class in for the tests to pass. Check with previous PRs that have migrated from XML -> Hibernate mappings.

**Comment by wikumChamith** [comment]
> There's some class you need to register the annotated class in for the tests to pass. Check with previous PRs that have migrated from XML -> Hibernate mappings.

@ibacher Thank you. Any reason why it only fails on GitHub builds?

**Comment by dkayiwa** [comment]
> Thank you. Any reason why it only fails on GitHub builds?

@wikumChamith you must be having something wrong with your local setup because this fails for me locally.

**Comment by wikumChamith** [comment]
> > Thank you. Any reason why it only fails on GitHub builds?
> 
> @wikumChamith you must be having something wrong with your local setup because this fails for me locally.

Does it still fails for you? Now build is completing on GitHub.

**Comment by dkayiwa** [comment]
@wikumChamith i mean that what you had claimed as NOT failing for you locally was failing for me locally. 😊

**Comment by wikumChamith** [comment]
> @wikumChamith i mean that what you had said as NOT failing for you locally was failing for me locally. 😊 

This is strange. I even ran that test individually and it passed for me.

File: api/src/main/java/org/openmrs/hl7/HL7InError.java [hunk 1]
```diff
@@ -9,24 +9,47 @@
...
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Lob;
+import javax.persistence.Table;
+
 /**
- * Represents a error in processing an hl7 message.
+ * Represents an error in processing a hl7 message.
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/hl7/HL7InError.java:26]: Shouldn't it be `an` hl7 message?

Comment (wikumChamith) [reply api/src/main/java/org/openmrs/hl7/HL7InError.java]: I updated this

**Pull request merged**